### PR TITLE
Remove warning messages from test files for `pytorch_distributed` and `chainermn` modules

### PR DESCRIPTION
**Pull request by nzw0301**
## Motivation

As a part of https://github.com/optuna/optuna/issues/3815, this PR remove all expected warning messages and resolve unnecessary avoidable warning from distributed integration: chainermn and pytorch_distributed.

## Description of the changes

As the title.

### CODE REVIEW

File: tests/integration_tests/test_chainermn.py
Comment (nzw0301) [inline tests/integration_tests/test_chainermn.py:467]: Minor simplification by using `ask` method. This way is also used in `tests/integration_tests/test_pytorch_distributed.py`.

File: tests/integration_tests/test_chainermn.py
Comment (nzw0301) [inline tests/integration_tests/test_chainermn.py:309]: Only rank 0 node shows `RuntimeWarning`, so I use the if branch.

**Comment by toshihikoyanase** [comment]
@keisuke-umezawa This PR is related to Chainer, so could you join the review, please?

File: tests/integration_tests/test_chainermn.py
Comment (toshihikoyanase) [reply tests/integration_tests/test_chainermn.py:309]: Could you tell me how do you select `warnings.catch_warnings` and `pytest.warns`. If we use `warnings.catch_warnings`, we may not need the condition branch.

**Review by toshihikoyanase — COMMENTED** [review]
Thank you for your PR. I have a question. Please take a look.

File: tests/integration_tests/test_chainermn.py
Comment (nzw0301) [reply tests/integration_tests/test_chainermn.py:309]: Since the other `test_suggest_*` methods use `with pytest.raises(ValueError)` such as https://github.com/optuna/optuna/blob/19a7fa5a48734978affecd54d501f61a8cfb4b94/tests/integration_tests/test_chainermn.py#L332-L334, I use `with pytest.warns(RuntimeWarning)` for consistency. But I suppose it doesn't look consistent with other methods, so I'm happy to use `warnings.catch_warnings` instead of the current `with pytest.raises(ValueError)`.

Comment (toshihikoyanase) [inline tests/integration_tests/test_chainermn.py:309]: > But I suppose it doesn't look consistent with other methods, so I'm happy to use `warnings.catch_warnings` instead of the current with `pytest.raises(ValueError)`.

I may not fully understand your opinion, but I also think that `warnings.catch_warnings` is simple .

```suggestion
                with warnings.catch_warnings():
                    warnings.simplefilter("ignore", category=RuntimeWarning)
                    mn_trial.suggest_float("x", low, high)
```

**Comment by nzw0301** [comment]
I realised `warnings.simplefilter("ignore", category=RuntimeWarning)` with `ValueError` cases would be unnecessary after merging https://github.com/optuna/optuna/pull/3914 into the master. Thus, we can wait for the merge of https://github.com/optuna/optuna/pull/3914 or send a follow-up to remove unnecessary `warnings.simplefilter("ignore", category=RuntimeWarning)` introduced by this PR.

**Comment by nzw0301** [comment]
I've simplified the changes in `chainermn` test in this PR thanks to the merge of #3914 as described in https://github.com/optuna/optuna/pull/3927#issuecomment-1236122920.

**Comment by toshihikoyanase** [comment]
Sorry for the delayed response. Let me check this PR today.

**Comment by nzw0301** [comment]
any update? :)

**Review by toshihikoyanase — APPROVED** [review]
I'm sorry for the delayed response. LGTM.

**Pull request merged**
